### Importing the necessary libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

# 1. You are a manager of a chinese restaurant. You want to determine whether the waiting time to place an order has changed in the past month from its previous population mean value of 4.5 minutes. State the null and alternative hypothesis.

The null hypothesis is that the population mean has not changed from its previous value of 4.5 minutes. This is stated 
as

H0: µ = 4.5

The alternative hypothesis is the opposite of the null hypothesis. Because the null hypothesis is that the population
mean is 4.5 minutes, the alternative hypothesis is that the population mean is not 4.5 minutes. This is stated as

H1 : µ not equal to 4.5

# 2. What is the p value if, in a two-tail hypothesis test, Zstat = +2.00?

In [2]:
# Calculating the p-value for Zstat=2
p_val=1-stats.norm.cdf(2)
p_val

0.02275013194817921

In [3]:
#since it is a two tailed test-
p_val*2

0.04550026389635842

p-value for Zstat =2 is 0.0455

# 3. Samy, Product Manager of K2 Jeans, wants to Launch a product line into a new market area.Survey of random sample of 400 households in that market showed a mean income per household of 30000 rupees. Standard deviation based on earlier pilot study of households is 8000 rupees. Samy strongly believes the product line will be adequately profitable only in markets where the mean household income is greater than 29000 rupees Samy wants our help in deciding whether the product line should be introduced in the New market Perform statistical analysis and based on that draw a conclusion.

Null Hypothesis: Mean income of household is less than or equal to 29000 

H0: µ <= 29000 

Alternate Hypothesis: Mean income of household is greater than 29000

H1: µ > 29000 

x̅= 30000

μ = 29000

σ= 8000

n = 400

𝑍=(x̅ − μ)/ (σ/(sqrt(n)) = (30000 – 29000) / (8000/sqrt(400))= 2.5

In [4]:
# Calculating the p-value for Zstat=2.5
1-stats.norm.cdf(2.5)

0.006209665325776159

Interpretation of p value: 
If the p-value is less than α(significance level), reject the null hypothesis.

Since α is not specified,we assume it to be 0.05. 

Since P-value (0.006) < α (0.05), we reject the null hypothesis and conclude that the mean income of household is greater than 29000

## Overview
Let's say you know the mean and standard deviation of a population. How can you tell if a sample is from this population or some other population? Although we may never know with 100% certainty, we can look for statistically significant differences between the sample statistics and the population paramters. This is done by first stating what is refered to as a null hypothesis, which in this scenario would be that there is no difference between the sample mean and the population mean. Then we look for statistical evidence to accept or reject the null hypothesis. What if we don't know the population parameters? 

## 4. Check whether the sample of Wine Scores is from the Wine Enthusiast Population. Let's assume the WineEnthusiast point scores are interval-scaled normally distributed data. (Assume 5% significance level: Zcritical value =  1.96) 

The population parameters are as follows:

Population mean (mu) = 88.44

Standard deviation (sigma) = 3.03

A sample of 10 wine scores yields a sample mean(x_bar) = 90.2

**a) Formulate the null hypothesis and alternative hypothesis**

* H<sub>0</sub>: The sample is from the population, x_bar = &mu;.
* H<sub>A</sub>: The sample is not from the population, x_bar != (not equal) &mu;.

**b) Calculate the test-statistics and conclude if the sample is typical to the population or not.**

In [5]:
z_critical = 1.96 # alpha level of 0.05 and two-tailed test
mu = 88.44
sigma = 3.03
x_bar = 90.2
N = 10
SE = sigma/np.sqrt(N)
z_stat = (x_bar - mu)/SE
print(z_stat)

1.8368345484806485


This is a two‐tailed test; so the 0.05 must be split such that 0.025 is in the upper tail and another 0.025 in the lower. 
The z‐value of the lower tail is –1.96. 
The Z- value that corresponds to the upper tail is 1.96. 

The null hypothesis of no difference will be rejected if the computed z statistic falls outside the range of –1.96 to 1.96.

Because 1.836 is between –1.96 and 1.96, the null hypothesis that the sample is from the population cannot be rejected. 

**c) Calculate the test-statistics and conclude if the sample is typical to the population or not if the sample size(N) is equal to 30.**

In [6]:
N = 30
SE = sigma/np.sqrt(N)
z_stat = (x_bar - mu)/SE
print(z_stat)

3.1814907630663205


Because 3.18 doesn’t lie between –1.96 and 1.96, the null hypothesis that the sample is from the population is rejected. That is, there is enough evidence that this sample can be considered different from population.

## 5. One-sample t- test 

The mass of a sample of N=20 acorns from a forest subjected to acid rain from a coal power plant are m = 8.8, 6.6, 9.5, 11.2, 10.2, 7.4, 8.0, 9.6, 9.9, 9.0, 7.6, 7.4, 10.4, 11.1, 8.5, 10.0, 11.6, 10.7, 10.3, and 7.0 g. 

Is the average mass of this sample different from the average mass of all acorns of &mu; = 10.0 g?

**a) Formulate the null and alternate hypothesis**

* H<sub>0</sub>: x&#772; - &mu; = 0, There is no difference between sample mean and the value of &mu;.
* H<sub>a</sub>: x&#772; - &mu; &ne; 0, There is difference between sample mean and the value of &mu;.

**b) Calculate the test-statistics and based on p-value provide a conclusion.**

In [7]:
x = [8.8, 6.6, 9.5, 11.2, 10.2, 7.4, 8.0, 9.6, 9.9, 9.0,
     7.6, 7.4, 10.4, 11.1, 8.5, 10.0, 11.6, 10.7, 10.3, 7.0]

mu = 10

t, p = stats.ttest_1samp(x, mu)
print("tstats = ", t, ", p-value = ", p)

tstats =  -2.2491611580763973 , p-value =  0.03655562279112415


Since P-value(0.03) < 0.05 (alpha) we reject the null hypothesis and conclude that the sample mean is not equal to the population mean or there is a difference between the sample mean and the value of &mu; Hence, average mass of this sample of acorns is not equal to &mu; = 10.

## 6. Independent (unpaired) two-sample t-test.
The mass of N<sub>1</sub>=20 acorns from oak trees up wind from a coal power plant and N<sub>2</sub>=30 acorns from oak trees down wind from the same coal power plant are measured. Is the mass of acorns from trees down wind different from the the ones from up wind? The sample sizes are not equal but we will assume that the population variance of sample 1 and sample 2 are equal.

#### sample up wind:
x1 = [10.8, 10.0, 8.2, 9.9, 11.6, 10.1, 11.3, 10.3, 10.7, 9.7, 
      7.8, 9.6, 9.7, 11.6, 10.3, 9.8, 12.3, 11.0, 10.4, 10.4]

#### sample down wind:
x2 = [7.8, 7.5, 9.5, 11.7, 8.1, 8.8, 8.8, 7.7, 9.7, 7.0, 
      9.0, 9.7, 11.3, 8.7, 8.8, 10.9, 10.3, 9.6, 8.4, 6.6,
      7.2, 7.6, 11.5, 6.6, 8.6, 10.5, 8.4, 8.5, 10.2, 9.2]


**a) Formulate null and alternate hypothesis.**

* Ho: x1 = x2 -> x1 - x2 = 0, There is no difference between the means
* Ha: x1 != x2 -> x1 - x2 != 0, There is difference between the means


**b) Calculate the test-statistics and based on p-value provide a conclusion.**

In [8]:
# sample up wind
x1 = [10.8, 10.0, 8.2, 9.9, 11.6, 10.1, 11.3, 10.3, 10.7, 9.7, 
      7.8, 9.6, 9.7, 11.6, 10.3, 9.8, 12.3, 11.0, 10.4, 10.4]

# sample down wind
x2 = [7.8, 7.5, 9.5, 11.7, 8.1, 8.8, 8.8, 7.7, 9.7, 7.0, 
      9.0, 9.7, 11.3, 8.7, 8.8, 10.9, 10.3, 9.6, 8.4, 6.6,
      7.2, 7.6, 11.5, 6.6, 8.6, 10.5, 8.4, 8.5, 10.2, 9.2]


t, p_value = stats.ttest_ind(x2, x1)
print("tstats = ",t, ", p_value = ", p_value)

tstats =  -3.5981947686898033 , p_value =  0.0007560337478801464


Since P-value(0.0007) < 0.05 (alpha) we reject the null hypothesis and conclude that there is a difference in the two means.

## 7. Paired samples t-test.
The average mass of acorns from the same N=30 trees down wind of a power plant is measured before (x<sub>1</sub>) and after (x<sub>2</sub>) the power plant converts from burning coal to buring natural gas. Does the mass of the acorns increase after the conversion from coal to natural gas? 

### sample before conversion to natural gas
x1 = np.array([10.8, 6.4, 8.3, 7.6, 11.4, 9.9, 10.6, 8.7, 8.1, 10.9,
      11.0, 11.8, 7.3, 9.6, 9.3, 9.9, 9.0, 9.5, 10.6, 10.3,
      8.8, 12.3, 8.9, 10.5, 11.6, 7.6, 8.9, 10.4, 10.2, 8.8])

### sample after conversion to natural gas
x2 = np.array([10.1, 6.9, 8.6, 8.8, 12.1, 11.3, 12.4, 9.3, 9.3, 10.8,
      12.4, 11.5, 7.4, 10.0, 11.1, 10.6, 9.4, 9.5, 10.0, 10.0,
      9.7, 13.5, 9.6, 11.6, 11.7, 7.9, 8.6, 10.8, 9.5, 9.6])



**a) Formulate null and alternate hypothesis.**

* H<sub>0</sub>: x2-x1 = 0 (The mean difference between the two samples is equal to zero)
* H<sub>1</sub>: x2-x1 != 0 (The mean difference between the two samples is not equal to zero)

**b) Calculate the test-statistics and based on p-value provide a conclusion.**

In [9]:
# sample before conversion to natural gas
x1 = np.array([10.8, 6.4, 8.3, 7.6, 11.4, 9.9, 10.6, 8.7, 8.1, 10.9,
      11.0, 11.8, 7.3, 9.6, 9.3, 9.9, 9.0, 9.5, 10.6, 10.3,
      8.8, 12.3, 8.9, 10.5, 11.6, 7.6, 8.9, 10.4, 10.2, 8.8])
# sample after conversion to natural gas
x2 = np.array([10.1, 6.9, 8.6, 8.8, 12.1, 11.3, 12.4, 9.3, 9.3, 10.8,
      12.4, 11.5, 7.4, 10.0, 11.1, 10.6, 9.4, 9.5, 10.0, 10.0,
      9.7, 13.5, 9.6, 11.6, 11.7, 7.9, 8.6, 10.8, 9.5, 9.6])


t, p = stats.ttest_1samp(x2-x1,0)
print("tstats = ", t, ", p-value = ", p) 

tstats =  3.905439081326491 , p-value =  0.0005168689824684378


Since P-value(0.0005) < 0.05 (alpha) we reject the null hypothesis and conclude that the mean difference(μd) between the two populations is not equal to zero and there is a significant difference between them.